shell/platform/embedder/embedder.h

```diff
@@ -1064,6 +1064,53 @@ typedef int64_t FlutterPlatformViewIdentifier;
 FLUTTER_EXPORT
 extern const int32_t kFlutterSemanticsNodeIdBatchEnd;
 
+typedef enum {
+  // Indicates the string should be announced character by character.
+  kSpellOut,
+  // Indicates the string should be announced using the specified locale.
+  kLocale,
+} FlutterStringAttributeType;
+
+// Indicates the assistive technology should announce out the string character
+// by character.
+//
+// See dart:ui's SpellOutStringAttribute.
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterSpellOutStringAttribute).
+  size_t struct_size;
+} FlutterSpellOutStringAttribute;
+
+// Indicates the assistive technology should announce the string using the
+// specified locale.
+//
+// See dart:ui's LocaleStringAttribute.
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterLocaleStringAttribute).
+  size_t struct_size;
+  // The locale of this attribute.
+  const char* locale;
```

Comment:
This string has to be statically allocated. Is that going to be a concern?

Reply:
This string is created dynamically ([here](https://github.com/flutter/engine/blob/f1f912c0c7c78a42248862396b06f796e457310f/lib/ui/semantics.dart#L684-L685) and [here](https://github.com/flutter/engine/blob/f1f912c0c7c78a42248862396b06f796e457310f/lib/ui/semantics/string_attribute.cc#L38C29-L53)).

What makes you say this has to be statically allocated? This mirrors other dynamically allocated strings like these:

https://github.com/flutter/engine/blob/f1f912c0c7c78a42248862396b06f796e457310f/shell/platform/embedder/embedder.h#L1108-L1118

Reply:
Technically doesn't have to be statically allocated but it does need to be valid for the entire duration that it's possible for it to be used by the embedder.

Typically, I'd this would match the duration of the corresponding node in the semantics tree, so as long as the string is part of a struct attached to the semantics tree handed to us via from dart:ui, I think you'll be alright. Typically when updating the embedder's semantics tree, we're making copies of these strings the moment we get the update callback, so realistically, it's just until that happens that this needs to remain valid.